File: 242-ValidAnagram/242-ValidAnagram.cpp
// Last updated: 2/11/2026, 11:08:23 AM
#include <unordered_map>
using namespace std;

class Solution {
public:
    bool isAnagram(string s, string t) {
        unordered_map<char,int> map_s;
        unordered_map<char,int> mapt;

        if (size(t)!= size(s)){
            return false;
        }
        for(int i = 0; i < size(s); ++i){
            auto it = map_s.find(s[i]);
            if (it != map_s.end()){
                map_s[s[i]] +=1;
            }else{
                map_s[s[i]] = 1;
            }
        }

        for(int i = 0; i < size(t); ++i){
            auto it = mapt.find(t[i]);
            if (it != mapt.end()){
                mapt[t[i]] +=1;
            }else{
                mapt[t[i]] = 1;
            }
        }

        for (auto &[key,value] :map_s){
            if (mapt.find(key) == mapt.end()){
                return false;
            }
            if(value != mapt[key]){
                return false;
            }
        }

        return true;
    }
};

File: 242-ValidAnagram/242-ValidAnagram.py
# Last updated: 2/11/2026, 11:08:25 AM
class Solution(object):
    def isAnagram(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        if len(s) != len(t):
            return False

        mp_s = {}
        mp_t = {}

        for ch in s:
            mp_s[ch] = mp_s.get(ch,0) + 1

        for ch in t:
            mp_t[ch] = mp_t.get(ch,0) + 1

        for key,value in mp_s.items():
            if mp_s[key] != mp_t.get(key,0):
                return False

        return True
        